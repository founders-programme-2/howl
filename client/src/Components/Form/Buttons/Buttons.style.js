import styled from 'styled-components';

const Submit = styled.input`
  font-family: 'Raleway', sans-serif;
  font-weight: 900;
  font-size: 1.5rem;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 20px;
  padding: 0.8rem;
  margin: 1rem;
  text-transform: uppercase;
  transition: all 0.2s ease-in-out;
  background: var(--highlight-color);
  color: white;
  &:hover {
    cursor: pointer;
    background: var(--main-bg-color);
    color: black;
    transform: scale(1.1);
  }
`;

const GoBack = styled.input`
  font-family: 'Raleway', sans-serif;
  font-weight: 900;
  font-size: 1.5rem;
  background: none;
  border: none;
  padding: 0.8rem;
  margin: 1rem;
  &:hover {
    cursor: pointer;
  }
`;

const ButtonsContainer = styled.div`
  margin: 2%;
  width: 40%;
  display: flex;
  justify-content: space-evenly;
`;

export { Submit, GoBack, ButtonsContainer };
