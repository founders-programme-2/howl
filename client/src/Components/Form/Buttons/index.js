import React from 'react';
import { Link } from 'react-router-dom';
import { Submit, GoBack, ButtonsContainer } from './Buttons.style';
import navigationUrls from '../../../constants/navigationUrls';

const { ADD_URL } = navigationUrls;

const Buttons = ({ handleSubmit }) => {
  return (
    <ButtonsContainer>
      <Link to={ADD_URL}>
        <GoBack type="button" value="Go Back" />
      </Link>
      <Submit type="button" value="Submit" onClick={handleSubmit} />
    </ButtonsContainer>
  );
};

export default Buttons;
