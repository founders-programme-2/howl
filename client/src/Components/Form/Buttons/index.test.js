import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import Buttons from './index';

test('Natural', () => {
  let state = {};
  const props = {
    handleUploadFile: event => {
      const img = new FormData();
      img.append('uploadedImg', event.target.files[0]);
      state = { ...state, uploadFlag: true, imageFlag: false };
      axios.post('/upload', img).then(({ data }) => {
        if (data.success) {
          this.setState({
            uploadFlag: false,
            imgLink: data.image,
            message: 'Image uploaded successfully!',
          });
        } else {
          this.setState({
            uploadFlag: false,
            imageFlag: true,
            imgLink: 'no url was retrieved',
            message: 'Something went wrong, please try again.',
          });
        }
      });
    },
  };

  const buttons = renderer
    .create(
      <BrowserRouter>
        <Buttons />
      </BrowserRouter>
    )
    .toJSON();
  expect(buttons).toMatchSnapshot();
});
