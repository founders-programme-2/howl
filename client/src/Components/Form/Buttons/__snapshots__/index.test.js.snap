// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
<div
  className="sc-htpNat hJFhOq"
>
  <a
    href="/add"
    onClick={[Function]}
  >
    <input
      className="sc-bwzfXH dIgSMj"
      type="button"
      value="Go Back"
    />
  </a>
  <input
    className="sc-bdVaJa dasZuf"
    type="button"
    value="Submit"
  />
</div>
`;
